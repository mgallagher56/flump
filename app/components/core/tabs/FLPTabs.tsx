import type { PropsWithChildren } from 'react';
import React from 'react';

import { Tabs } from '@chakra-ui/react';

interface FLPTabsProps {
  orientation?: 'horizontal' | 'vertical';
}

const FLPTabs: React.FC<PropsWithChildren<FLPTabsProps>> = (props) => {
  return <Tabs {...props}>{props.children}</Tabs>;
};

export default FLPTabs;
