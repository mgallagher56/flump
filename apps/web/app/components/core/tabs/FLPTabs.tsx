import { type FC, type PropsWithChildren } from 'react';

import { Tabs, type TabsRootProps } from '@chakra-ui/react/tabs';

import type { TabData } from './types';

interface FLPTabsProps extends Omit<TabsRootProps, 'children'> {
  data: TabData[];
  orientation?: 'horizontal' | 'vertical';
}

const FLPTabs: FC<PropsWithChildren<FLPTabsProps>> = ({ data, orientation }) => {
  const FLPOrientation = orientation ?? 'horizontal';
  return (
    <Tabs.Root defaultValue={data[0].value} orientation={FLPOrientation}>
      <Tabs.List>
        {data.map((item) => (
          <Tabs.Trigger key={item.value} disabled={item.disabled} value={item.value}>
            {item.label}
          </Tabs.Trigger>
        ))}
      </Tabs.List>
      {data.map((item) => (
        <Tabs.Content key={item.value} value={item.value}>
          {item.children}
        </Tabs.Content>
      ))}
    </Tabs.Root>
  );
};

export default FLPTabs;
