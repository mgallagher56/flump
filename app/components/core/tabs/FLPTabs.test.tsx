import React from 'react';

import { render } from '@testing-library/react';
import { describe, expect, test } from 'vitest';

import FLPTabs from './FLPTabs';

describe('FLPTabs', () => {
  test('renders correctly', () => {
    const { container } = render(
      <FLPTabs>
        <div>Tab 1</div>
        <div>Tab 2</div>
        <div>Tab 3</div>
      </FLPTabs>
    );
    expect(container).toMatchSnapshot();
  });
});
