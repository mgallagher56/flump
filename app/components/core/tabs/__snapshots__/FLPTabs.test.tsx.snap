// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FLPTabs > renders correctly 1`] = `
<div>
  <div
    class="chakra-tabs css-0"
  >
    <div>
      Tab 1
    </div>
    <div>
      Tab 2
    </div>
    <div>
      Tab 3
    </div>
  </div>
</div>
`;
